iOS-Weather-App/AppDelegate.swift

```diff
@@ -10,27 +10,38 @@ import UIKit
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
 
-
+    var window: UIWindow?
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        // Override point for customization after application launch.
+        
+        window = UIWindow(frame: UIScreen.main.bounds)
+        window?.makeKeyAndVisible()
+        
+        let weatherController = WeatherViewController()
+        let navigationController = UINavigationController(rootViewController: weatherController)
```

Comment:
let's create a navigationController as a global var like a window property, and you wouldn't send navController to the functions